core/tx_pool.go

```diff
@@ -1031,6 +1044,14 @@ func (pool *TxPool) demoteUnexecutables(state *state.StateDB) {
 			log.Trace("Demoting pending transaction", "hash", hash)
 			pool.enqueueTx(hash, tx)
 		}
+		// If there's a gap in front, warn (should never happen) and postpone all transactions
+		if list.Len() > 0 && list.txs.Get(nonce) == nil {
+			for _, tx := range list.Cap(0) {
+				hash := tx.Hash()
+				log.Warn("Demoting invalidated transaction", "hash", hash)
```

Comment:
I think all 'should never happen' errors should be level ERROR.

Reply:
Agreed, will fix

Reply:
Fixed, ptal